fedimint-api-client/src/api/ws.rs

```diff
@@ -18,127 +17,94 @@ use jsonrpsee_wasm_client::{Client as WsClient, WasmClientBuilder as WsClientBui
-                    }
-                }
+        trace!(target: LOG_NET_WS, %url, "Creating new websocket connection");
+
+        #[cfg(not(target_family = "wasm"))]
+        let mut client = {
+            use jsonrpsee_ws_client::{CustomCertStore, WsClientBuilder};
+            use tokio_rustls::rustls::RootCertStore;
+
+            install_crypto_provider().await;
+            let webpki_roots = webpki_roots::TLS_SERVER_ROOTS.iter().cloned();
+            let mut root_certs = RootCertStore::empty();
+            root_certs.extend(webpki_roots);
+
+            let tls_cfg = CustomCertStore::builder()
+                .with_root_certificates(root_certs)
+                .with_no_client_auth();
+
+            WsClientBuilder::default()
+                .max_concurrent_requests(u16::MAX as usize)
+                .with_custom_cert_store(tls_cfg)
+        };
+
+        #[cfg(target_family = "wasm")]
+        let client = WsClientBuilder::default().max_concurrent_requests(u16::MAX as usize);
+
+        if let Some(api_secret) = api_secret {
+            #[cfg(not(target_family = "wasm"))]
+            {
+                // on native platforms, jsonrpsee-client ignores `user:pass@...` in the Url,
+                // but we can set up the headers manually
+
+                use base64::Engine as _;
+                use jsonrpsee_ws_client::{HeaderMap, HeaderValue};
+                let mut headers = HeaderMap::new();
+
+                let auth = base64::engine::general_purpose::STANDARD
+                    .encode(format!("fedimint:{api_secret}"));
+
+                headers.insert(
+                    "Authorization",
+                    HeaderValue::from_str(&format!("Basic {auth}")).expect("Can't fail"),
+                );
+
+                client = client.set_headers(headers);
+            }
+            #[cfg(target_family = "wasm")]
+            {
+                // on wasm, url will be handled by the browser, which should take care of
+                // `user:pass@...`
+                let mut url = url.clone();
+                url.set_username("fedimint")
+                    .map_err(|_| PeerError::InvalidEndpoint(anyhow!("invalid username")))?;
+                url.set_password(Some(&api_secret))
+                    .map_err(|_| PeerError::InvalidEndpoint(anyhow!("invalid secret")))?;
 
                 let client = client
                     .build(url.as_str())
                     .await
                     .map_err(|err| PeerError::InternalClientError(err.into()))?;
 
-                Ok(Arc::new(client))
-            })
-            .await?;
+                return Ok(Arc::new(client));
+            }
+        }
 
-        trace!(target: LOG_NET_WS, %url, "Using websocket connection");
-        Ok(conn.clone())
+        let client = client
+            .build(url.as_str())
+            .await
+            .map_err(|err| PeerError::InternalClientError(err.into()))?;
+
+        Ok(Arc::new(client))
```

Comment:
nit: would have been a nice opportunity to split that fn a bit (client init, secret to header conversion)